### PR TITLE
IPv6/multi : An adapted Zynq driver for multiple EMACs

### PR DESCRIPTION
**Pull request by htibosch**
Description
-----------
This PR make the Zynq driver compatible with the IPv6/multi branch.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: portable/NetworkInterface/Zynq/NetworkInterface.c [hunk 1]
```diff
@@ -1,6 +1,6 @@
 /*
- * FreeRTOS V202002.00
- * Copyright (C) 2020 Amazon.com, Inc. or its affiliates.  All Rights Reserved.
+ * FreeRTOS+FAT V2.3.3
```
Comment (AniruddhaKanhere) [inline portable/NetworkInterface/Zynq/NetworkInterface.c:2]: `+TCP`

File: portable/NetworkInterface/Zynq/NetworkInterface.c [hunk 2]
```diff
@@ -88,6 +92,16 @@
     #define configEMAC_TASK_STACK_SIZE    ( 8 * configMINIMAL_STACK_SIZE )
 #endif
 
+#if ( ipconfigNIC_LINKSPEED100 != 1 )
+
+/* When the PHY is forces to work with a speed of 100 Mbps
+ * many outgoing packets seem to get dropped.
+ */
+    #warning ipconfigNIC_LINKSPEED100 is btoken. Are you sure?
```
Comment (AniruddhaKanhere) [inline portable/NetworkInterface/Zynq/NetworkInterface.c:100]: `broken`

**Review by AniruddhaKanhere — CHANGES_REQUESTED** [review]
I am yet to review the whole PR. But these are some minor things that I found

File: portable/NetworkInterface/Zynq/NetworkInterface.c (hunk 2 above)
Comment (htibosch) [reply portable/NetworkInterface/Zynq/NetworkInterface.c:100]: I changed both, thank you.

**Review by AniruddhaKanhere — DISMISSED** [review]

File: portable/NetworkInterface/Zynq/NetworkInterface.c (hunk 1 above)
Comment (aggarg) [reply portable/NetworkInterface/Zynq/NetworkInterface.c:2]: Still not changed?

File: portable/NetworkInterface/Zynq/NetworkInterface.c (hunk 2 above)
Comment (aggarg) [reply portable/NetworkInterface/Zynq/NetworkInterface.c:100]: Doesn't seem changed?

File: portable/NetworkInterface/Zynq/uncached_memory.c [hunk 3]
```diff
@@ -1,6 +1,6 @@
 /*
- * FreeRTOS V202002.00
- * Copyright (C) 2020 Amazon.com, Inc. or its affiliates.  All Rights Reserved.
+ * FreeRTOS+FAT V2.3.3
```
Comment (aggarg) [inline portable/NetworkInterface/Zynq/uncached_memory.c:2]: +TCP here too?

File: portable/NetworkInterface/Zynq/uncached_memory.h [hunk 4]
```diff
@@ -1,3 +1,29 @@
+/*
+ * FreeRTOS+FAT V2.3.3
```
Comment (aggarg) [inline portable/NetworkInterface/Zynq/uncached_memory.h:2]: +TCP

Comment (htibosch) [reply portable/NetworkInterface/Zynq/uncached_memory.h]: Indeed, I had committed the changes but not pushed them. Sorry for that.
Hein

**Review by n9wxu — APPROVED** [review]

**Comment by AnandPavaskar** [comment]
I am trying to port FreeRTOS/TCP IP multi stack for the Xilinx  Zync (dual EMAC) and ending up with lot of debugging , is there a demo port already avilable. If yes can a git link be sent pls.
This will save me some time and help me do my Token Bucket  algorithim directly.

**Comment by htibosch** [comment]
Hi @AnandPavaskar,

You wrote:
> I am trying to port FreeRTOS/TCP IP multi stack for the Xilinx Zync (dual EMAC) and ending up with lot of debugging , is there a demo port already avilable. If yes can a git link be sent pls.

I have a Zynq project in a github repo called [freertos_plus_projects](https://github.com/htibosch/freertos_plus_projects/tree/master/plus/xilinx).

Unfortunately I have simple boards like the Zybo and MicroZed, which only have a single EMAC.

Earlier, there were developers who managed to use two EMACs. He had to get into the details of the FPGA to enabled both EMACs and connect them properly.

If you have more questions, maybe it is better if you open a post on the [FreeRTOS forum](https://forums.freertos.org), the audience will be much bigger than here on github.

> This will save me some time and help me do my Token Bucket algorithm directly.

The FreeRTOS forum is the best place to ask.

**Comment by AnandPavaskar** [comment]
Many thanks @htibosch I also  posted on FreeRTOS forum and seen your reply.
Hope to get this working soon :).

**Pull request merged**